### PR TITLE
🐛Don't expand in array if not string

### LINKED ISSUE #27414: Error: null is not an object (evaluating 'a.length')
This is an error that started appearing in [AMP Error Reporting](http://go/ampe/CIj6jN2ox8jYBg) Feb 18, 2020 in [2003101714470](https://github.com/ampproject/amphtml/releases/tag/2003101714470)

```
Error: null is not an object (evaluating 'a.length')
    at length (https://raw.githubusercontent.com/ampproject/amphtml/2003101714470/src/service/url-expander/expander.js:75)
    at source (https://raw.githubusercontent.com/ampproject/amphtml/2003101714470/src/service/url-replacements-impl.js:851)
    at item (https://raw.githubusercontent.com/ampproject/amphtml/2003101714470/extensions/amp-analytics/0.1/variables.js:345)
    at (https://raw.githubusercontent.com/ampproject/amphtml/2003101714470/extensions/amp-analytics/0.1/variables.js:344)
```

/tryassign @micajuine-ho as the last person to touch `expandTemplate` and /tryassign @calebcordry  as the last person to touch `expand`. Note that none of the files in this stacktrace seem to have been altered recently, so I'm just making a best guess on the assignees. Feel free to reassign.

### PR DESCRIPTION
**Pull request by micajuine-ho**
Closes #27414

### CODE REVIEW

File: extensions/amp-analytics/0.1/test/test-variables.js [hunk 1]
```diff
@@ -214,7 +214,7 @@ describes.fakeWin('amp-analytics.VariableService', {amp: true}, (env) => {
     });
 
     it('handles array with no vars', () => {
-      check('${array}', 'foo,bar', {
+      return check('${array}', 'foo,bar', {
```
Comment (zhouyx) [inline extensions/amp-analytics/0.1/test/test-variables.js:217]: There are maybe other places where we fail to return the `check()` promise, could you double check?

Comment (zhouyx) [reply extensions/amp-analytics/0.1/test/test-variables.js]: https://github.com/ampproject/amphtml/blob/master/extensions/amp-analytics/0.1/test/test-variables.js#L121

Comment (zhouyx) [reply extensions/amp-analytics/0.1/test/test-variables.js]: https://github.com/ampproject/amphtml/blob/master/extensions/amp-analytics/0.1/test/test-variables.js#L298

**Comment by micajuine-ho** [comment]
Discussed offline, added test to catch this bug (previous test didn't check for null)

Will consolidate code in next PR

**Pull request merged**